contracts/providers/IdleController.sol

```diff
@@ -0,0 +1,161 @@
+      underlyingDecimals = ERC20(uToken).decimals();
+      setBondModel(bondModel_);
+    }
+
+    function _beforeCTokenBalanceChange() external override onlyPool {}
+
+    function _afterCTokenBalanceChange(uint256 prevCTokenBalance_) external override onlyPool {
+        // at this point compound.finance state is updated since the pool did a deposit or withdrawl just before, so no need to ping
+        updateCumulativesInternal(prevCTokenBalance_, false);
+        IYieldOracle(oracle).update();
+    }
+
+    function updateCumulativesInternal(uint256 val, bool pingIdle_) internal {
+        uint256 timeElapsed = block.timestamp - prevCumulationTime;
+
+        // only cumulate once per block
+        if (0 == timeElapsed) {
+            return;
+        }
+
+        uint256 exchangeRateStoredNow = IIdleToken(cToken).tokenPriceWithFee(address(pool));
+
+        if (prevExchangeRateCurrent > 0) {
+          // cumulate a new supplyRate delta: cumulativeSupplyRate += (cToken.exchangeRateCurrent() - prevExchnageRateCurrent) / prevExchnageRateCurrent
+          // cumulativeSupplyRate eventually overflows, but that's ok due to the way it's used in the oracle
+          cumulativeSupplyRate += exchangeRateStoredNow.sub(prevExchangeRateCurrent).mul(EXP_SCALE).div(prevExchangeRateCurrent);
+        }
+
+        prevCumulationTime = block.timestamp;
+
+        // exchangeRateStored can increase multiple times per block
+        prevExchangeRateCurrent = exchangeRateStoredNow;
+    }
+
+    function providerRatePerDay() public override returns (uint256) {
+        if (IYieldOracle(oracle).consult(1 days) == 0) {
+            return MathUtils.min(BOND_MAX_RATE_PER_DAY, spotDailyRate());
+        }
+        return MathUtils.min(
+            MathUtils.min(BOND_MAX_RATE_PER_DAY, spotDailyRate()),
+            IYieldOracle(oracle).consult(1 days)
+        );
+    }
+
+    function cumulatives() public override returns (uint256) {
+        uint256 timeElapsed = block.timestamp - prevCumulationTime;
+        // only cumulate once per block
+        if (0 == timeElapsed) {
+          return cumulativeSupplyRate;
+        }
+        uint256 cTokenBalance = IdleProvider(pool).cTokenBalance();
+        updateCumulativesInternal(cTokenBalance, false);
+        return cumulativeSupplyRate;
+    }
+
+    function cTokensToUnderlying(uint256 cTokens_, uint256 exchangeRate_) public pure returns (uint256) {
+      return cTokens_.mul(exchangeRate_).div(EXP_SCALE);
+    }
+
+    function harvest(uint256)
+      public
+      returns (address[] memory tokens, uint256[] memory rewardAmounts, uint256 underlyingHarvestReward)
+    {
+        (address[] memory tokens, uint256[] memory rewardTotal, uint256[] memory rewardSold)
+          = IdleProvider(pool).claimRewardsTo(MAX_UINT256, rewardsCollector);
+        emit Harvest(msg.sender, tokens, rewardTotal, rewardSold, 0, HARVEST_COST);
+        return (tokens, rewardTotal, 0);
+    }
```

Comment:
Ideally your `harvest` method will be compatible with the rest of the controllers.